sopel/config/__init__.py

```diff
@@ -47,6 +47,16 @@ def __str__(self):
         return 'ConfigurationError: %s' % self.value
 
 
+class NotFound(ConfigurationError):
```

Comment:
As discussed on IRC:

* let's go for `sopel.config.ConfigNotFound`,
* we agree that `something.SomethingSomething` is not OK,
* but `something.SomethingNotFound` is (because it's an exception),

which is a good compromise between clarity and avoiding repetition.